src/main/java/Graph.java

```diff
@@ -0,0 +1,99 @@
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+/**
+ * An undirected and unweighted graph. A graph is defined by a set of vertices V and a set of edges E. A single
+ * vertex is some arbitrary element often represented as a "node" and an edge is a pair of vertices from the set V
+ * indicating some relationship between the vertices. Nodes are considered "neighbours" if they have an edge between
+ * them. Consider the below example:
+ * <p>
+ * a - b - d
+ * \ /
+ * c
+ * <p>
+ * V = {a, b, c, d}
+ * E = {(a, b), (a, c), (b, c), (b, d)}
+ *
+ * @param <T> The "node" objects
+ */
+public interface Graph<T> {
+
+    /**
+     * Adds the specified node to the graph. If the node already exists, the graph remains unchanged.
+     *
+     * @param node The node to be added to the graph
+     * @return True if the node was added successfully, false otherwise
+     */
+    boolean addNode(T node);
+
+    /**
+     * Removes the specified node from the graph. All edges incident on the node are also removed.
+     *
+     * @param node The node to be removed from the graph
+     * @return True if the node was removed successfully
+     * @throws NoSuchElementException If the specified node does not exist within the graph
```

Comment:
exception here seems nonsensical if the return value already encodes that information

Reply:
The idea was to follow what was done with the other data structures in the previous course, but upon looking at the Java docs, I see that what was done with the previous data structures was perhaps atypical. For example, the `Bag` data structures throws an exception if removing a nonexistent element. Checking [Java's list ](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/List.html), which was the basis of the bag data structure, does not throw an exception. 

My guess is that I followed the removes from the stack and queue (pop, dequeue) that threw an exception when empty, but obviously that's a different case. 

Long story short, good call; I'll make the change here and will man an issue for 102's content to update the removes to NOT throw an exception when the element does not exist.